pkg/table/tables/index.go

```diff
@@ -427,7 +434,9 @@ func (c *index) Delete(ctx table.MutateContext, txn kv.Transaction, indexedValue
 						if err != nil {
 							return err
 						}
-						if !h.Equal(oh) {
+						// The handle passed in may be a `PartitionHandle`,
+						// so we can't directly do comparation with them.
+						if !handleEqual(h, oh) {
```

Comment:
If the passed handle is PartaitionHandle, why oh isn't partitionHandle here?

Reply:
Take the case in #57414 as example, the handle `h` is retrieved from `FetchDuplicatedHandle`. Since global index is first checked and duplication is found, we will get `PartitionHandle`.
https://github.com/pingcap/tidb/blob/043f833c4985ee489948bfb75cd1b3256cc7bbd6/pkg/executor/insert.go#L272-L273
But the newly added index is not global index, so `oh` is just `IntHandle`.

In addition, there may also be opposite situations that `oh` is `PartitionHandle` and `h` is `IntHandle`. But `IntHandle` can be compared with `PartitionHandle` correctly.